ge/org.osate.ge/src/org/osate/ge/internal/ui/editor/AgeContentOutlinePage.java

```diff
@@ -484,6 +491,7 @@ private void selectEditorDiagramElements() {
 		if (getTreeViewer() != null && getTreeViewer().getContentProvider() != null
 				&& !outlineElements.equals(editorElements)) {
 			editor.selectDiagramNodes(outlineElements);
+			editor.reveal(outlineElements.iterator().next());
```

Comment:
This assumes that outline nodes has at least one item selected. Will throw an exception if there is ever a case where the selection is empty. Consider outlineNodes.stream().findAny().ifPresent(...);

Reply:
Yeah, probably a  good idea.  I used findFirst since findAny has no guarantee of being the first element